src/appleDeviceUtils/appleDeviceUtils.js

```diff
@@ -0,0 +1,105 @@
+// Do not remove -- Apple devices change this when they support modern webkit messaging
+let hasModernWebkitAPI = false
+// INJECT hasModernWebkitAPI HERE
+
+// The native layer will inject a randomised secret here and use it to verify the origin
+const secret = 'PLACEHOLDER_SECRET'
+
+const ddgGlobals = window.navigator.ddgGlobals
+
+/**
+ * Sends message to the webkit layer
+ * @param {String} handler
+ * @param {*} data
+ * @returns {*}
+ */
+const wkSend = (handler, data = {}) =>
+    window.webkit.messageHandlers[handler].postMessage(data)
+
+/**
+ *
+ * @param {String} randomMethodName
+ * @param {Function} callback
+ */
+const generateRandomMethod = (randomMethodName, callback) => {
```

Comment:
I've moved decryption out of the randomised method and in the application logic, this way, even if the attacker gets ahold of the method it only ever receives encrypted data.